A1-http-connect-proxy-support.md

```diff
@@ -1,222 +1,265 @@
+        CONNECT handshaker.
+    -   Right before we connect to the target address, the proxy mapper can be
+        used to programmatically override the address that we will connect to.
+        It will also be able to set the internal state to trigger use of the
+        HTTP CONNECT handshaker.
 
 ### Addressing Use Cases
 
-This new functionality will be used to address the three use-cases
-described above, as follows.
+This new functionality will be used to address the three use-cases described
+above, as follows.
 
 #### Case 1
 
-Case 1 will be addressed using the proxy mapper hook for overriding the
-server name to resolve.  Sites that may want to send only external servers
-through the proxy can implement their own proxy mapper to do this.
-We will provide a default proxy mapper implementation that looks for the
-existence of the `http_proxy` environment variable (or equavalent
-signal).  If that signal is present, it will do the following:
-- Ask the resolver to resolve the proxy name instead of the server name
-  requested via the client API.  (Note that this will cause us to
-  create a single subchannel to the proxy, as opposed to one
-  subchannel for each backend server.)
-- Add the state described above triggering the use of the HTTP CONNECT
-  handshaker, with the argument set to the server name requested via
-  the client API.
+Case 1 will be addressed using the proxy mapper hook for overriding the server
+name to resolve. Sites that may want to send only external servers through the
+proxy can implement their own proxy mapper to do this. We will provide a default
+proxy mapper implementation that looks for the existence of the `http_proxy`
+environment variable (or equavalent signal). If that signal is present, it will
+do the following: - Ask the resolver to resolve the proxy name instead of the
+server name requested via the client API. (Note that this will cause us to
+create a single subchannel to the proxy, as opposed to one subchannel for each
+backend server.) - Add the state described above triggering the use of the HTTP
+CONNECT handshaker, with the argument set to the server name requested via the
+client API.
 
 Note that in this case, because the client cannot know the set of server
-addresses, it is impossible to use the normal gRPC client-side per-call
-load balancing.  It *is* possible to do load balancing on a per-connection
-basis, but that may not spread out the load evenly if different clients
-impose a different amount of load.  (Note that even if we established
-multiple connections to the proxy, we would have no guarantee that each
-connection would go to a different backend server.)
+addresses, it is impossible to use the normal gRPC client-side per-call load
+balancing. It *is* possible to do load balancing on a per-connection basis, but
+that may not spread out the load evenly if different clients impose a different
+amount of load. (Note that even if we established multiple connections to the
```

Comment:
The wording "...even if we established multiple connections to the..." implies that we currently don't do this which is probably not the intent. Basically with the current implementation (as described in the previous lines 122-124) the gRPC client will create a separate connection to the same proxy when asked to create a channel to the target.

It is important to clarify this point for the use-case we have been discussing and for which this PR has been created to explain the functioning.

Reply:
Not proceeding with changes to this doc anymore